third-party/realdds/doc/streaming.md

```diff
@@ -1,2 +1,159 @@
+
+
+### Video
+
+This is a simplified IDL for reference:
+
+```rosidl
+struct Time {
+    long              sec;
+    unsigned long     nanosec;
+};
+
+struct Header {
+    Time              stamp;
+    string            frame_id;
+};
+
+struct Image {
+    Header            header;
+    unsigned long     height;
+    unsigned long     width;
+    string            encoding;
+    octet             is_bigendian;   // false
+    unsigned long     step;
+    sequence< octet > data;
+};
+```
+
+The `encoding` is the same as the currently set profile format, and shouldn't change between frames. Neither should the `width`, `height`, `step`, or `frame_id`.
+
+
+### Motion
+
+This is a simplified IDL for reference:
+
+```rosidl
+struct Vector3 {
+    double x;
+    double y;
+    double z;
+};
+
+struct Imu {
+    Header            header;
+    Quaternion        orientation;
+    double[9]         orientation_covariance;
+    Vector3           angular_velocity;
+    double[9]         angular_velocity_covariance;
+    Vector3           linear_acceleration;
+    double[9]         linear_acceleration_covariance;
+};
+```
+
+
+## Metadata
+
+There are no frame numbers in the ROS2 formats.
+
+It is therefore impossible to communicate a frame number unless additional data is used. This is what [metadata](metadata.md) is for. Without metadata (which is optional), frames will not be numbered, or will be up to the client to number (as happens in librealsense).
+
+The same is true for any frame information that the server generates.
+
+
+## Controls
+
+
+#### `open-streams`
+
+This control tries to configure the profile(s) used for streaming.
+
+Note that it does not start streaming. It is a request. It may fail.
+
+Notably, if a sensor is already streaming, this command will likely fail though the behavior is up to the server.
+
+```JSON
+{
+    "id": "open-streams",
+    "stream-profiles": {
+        "Color": [30,"rgb8",1280,720]
+    }
+}
+```
+
+The `stream-profiles` is a mapping from `stream-name` to a `profile`. The `profile` is an array of `[frequency, format, width, height]`, similar to those used in initialization.
+
+If the `stream-name` or `profile` are not found, this should result in an error.
+
+If a field called `reset` is set to `true` (the default -- the above JSON therefore has it at `true`), any previous `open-streams` are forgotten and the server resets all sensors to default profiles before processing the `stream-profiles` (which can be empty).
+
+Multiple streams can be configured by a single `open-streams`. If these streams share a `sensor-name` (see [initialization](initialization.md#stream-header)), they must be compatible or an error will result.
+
+If the stream profile cannot be changed (sensor is already open, for example), an error will be the result. This usually means that, once the stream is streaming, its profile cannot be modified by anyone. The only way is to stop all subscriptions first.
```

Comment:
I would use `configured` instead of `changed`.
Also sensor is `open` might be problematic because you `reset` open sensors. I would use `streaming` but it's repetitive with the next line. Maybe just remove the parenthesized part.